client/src/components/rapid/Rapid.js

```diff
@@ -4,43 +4,48 @@ import PropTypes from "prop-types";
 import Image from "react-bootstrap/Image";
 
 import NextRapid from "./NextRapid";
-import Features from "./Features";
+import Features from "../shared/Features";
 import RapidHeader from "./RapidHeader";
+import store from "../../rematch/store";
 
 /**
  * The Rapid container that holds all components related to an individual rapid
  */
 
-const Rapid = ({ rapid, river, level }) => {
+const Rapid = ({ riverId, sectionId, rapid, section, level }) => {
+  const { dispatch } = store;
+
   const arrowArray = rapid.arrows.map((arrow, key) => (
     <NextRapid
       linkId={arrow.linkId}
       bottom={arrow.bottom}
       right={arrow.right}
       key={`arrow${key}`}
-      river={river}
+      section={section}
     />
   ));
 
   return (
     <div className="Rapid">
       <Image
         className="basemap"
-        src={`https://bent-shaft-maps.s3.amazonaws.com/maps/river_W41oYWjV4/section_jswy1FCZu/${rapid.id}.jpg`}
+        src={`https://bent-shaft-maps.s3.amazonaws.com/maps/${riverId}/${sectionId}/${rapid.id}.jpg`}
         alt="River Map"
       />
-      <Features level={level} rapid={rapid} />
+      <Features
+        level={level}
+        rapid={rapid}
+        reducers={dispatch.testEnvironment}
```

Comment:
I would import the `store` inside the `Features.js`.
Now reducers are traveling through props 1 extra time.

Reply:
Based on how I currently have it set up, you are right. 

What I was thinking for the future, is that `<Features />` is called in `<Rapid />` for when its used in production, and `<TestEnvironment />` for when its used in the development test environment. When the reducers are passed into <Features /> from those components, we can give `<Features />` its own individual set of `reducers` (because I test-environment reducers will differ slightly from production reducers). Right now Im passing `dispatch.testEnvironment` to `<Rapid />`, but this is going to be `dispatch.someOtherSetOfReducers` once we build them.